File: Data Structures & Algorithms/trapping-rain-water/submission-1.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        //maxleft height nikalenge
        vector<int>maxleftarr;
            int maxleft=0;
        for(int i=0;i<height.size();i++){
            int ele=height[i];
            maxleft=max(maxleft,ele);
            maxleftarr.push_back(maxleft);
        }
        //maxright height nikalenge
        vector<int>maxrightarr(height.size());
            int maxright=0;
        for(int i=height.size()-1;i>=0;i--){
            int ele=height[i];
            maxright=max(maxright,ele);
            maxrightarr[i]=maxright;

        }
        

        // 
        int total=0;
        for(int i=0;i<height.size();i++){
            int maxheight=min(maxleftarr[i],maxrightarr[i])-height[i];
            total+=maxheight;
        }
        return total;


    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0
        r=0
        st=set()
        maxlen=0

        while r<len(s):

            while s[r] in st:
                st.remove(s[l])
                l+=1
            
            st.add(s[r])
            maxlen= max(maxlen,r-l+1)
            r+=1

        return maxlen    

